flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java

```diff
@@ -419,12 +420,18 @@ public static FlinkPod loadPodFromTemplateFile(
         final List<Container> otherContainers = new ArrayList<>();
         Container mainContainer = null;
 
-        for (Container container : pod.getInternalResource().getSpec().getContainers()) {
-            if (mainContainerName.equals(container.getName())) {
-                mainContainer = container;
-            } else {
-                otherContainers.add(container);
+        if (null != pod.getInternalResource().getSpec()) {
+            for (Container container : pod.getInternalResource().getSpec().getContainers()) {
+                if (mainContainerName.equals(container.getName())) {
+                    mainContainer = container;
+                } else {
+                    otherContainers.add(container);
+                }
             }
+            pod.getInternalResource().getSpec().setContainers(otherContainers);
+        } else {
+            // Set empty spec for taskmanager pod template
```

Comment:
This is not only for taskmanger pod template.

Reply:
Right, it's also called by jobmanager pod template, but when no spec field on template, otherContainers is empty, and the PodSpec containers initial is an empty `List<Containers>` ,So there is no need to set an empty otherContainers to pod Spec.
PodSpec containers property:
```
    @JsonProperty("containers")
    private List<Container> containers = new ArrayList();
```

Reply:
Fixed.